### DISCUSSION TITLE
How do I reset a guage after collection on FAST API?

### DISCUSSION THREAD

**Discussion by denson-abraham** [prometheus/prometheus#14743] [Q&A]
So I have a fast API application which has a llm endpoint. So I count tokens everytime that llm is called and I use gauge to capture the usage trend.

Now if I call the API  /llm    at 12:00 AM and it consumed 100 tokens and my gauge value is 100, prometheus collects it, it is good. however my next request is not hit till let's say 5 minutes. For the next 5 minutes, it keeps showing that my token consumed is 100 where as after 12 AM there were no token called, I want it to be zero.   Is there any solution to this. I am really stuck here.
Also /llm is called rarely so I need to keep accurate count. 

So in short, once my gauge value is collected, it should be reset to 0 until the new value is is set. Right now the value remains the same until a new request is hit. Is it possible to get it done?
